### PR TITLE
feat(kit): new component `ProgressSegmented`

### PR DESCRIPTION
**Pull request by nsbarsukov**
## PR Checklist

Please check if your PR fulfills the following requirements:

-   [X] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0-beta.4/)
-   [ ] Tests for the changes have been added (for bug fixes / features)
-   [X] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

-   [ ] Bugfix
-   [X] Feature
-   [ ] Refactoring (no functional changes, no api changes)
-   [ ] Other... Please describe:

## What is the current behavior?

Closes #

## Does this PR introduce a breaking change?

-   [ ] Yes
-   [X] No

### CODE REVIEW

File: projects/demo/src/modules/app/app.routes.ts [hunk 1]
```diff
@@ -832,6 +832,16 @@ export const ROUTES = [
             title: 'ProgressCircle',
         },
     },
+    {
+        path: 'components/progress-segmented',
+        loadChildren: () =>
+            import(
+                `projects/demo/src/modules/components/progress-segmented/progress-segmented.module`
```
Comment (waterplea) [inline projects/demo/src/modules/app/app.routes.ts:839]: Shorten path please

File: projects/kit/components/progress/progress-segmented/progress-segmented.component.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import {ChangeDetectionStrategy, Component, HostBinding, Input} from '@angular/core';
+import {tuiDefaultProp} from '@taiga-ui/cdk';
+import {TuiSizeS} from '@taiga-ui/core';
+
+export const nonNegativeInt = (value: number | string) =>
+    Number.isInteger(+value) && value >= 0;
```
Comment (waterplea) [inline projects/kit/components/progress/progress-segmented/progress-segmented.component.ts:6]: `+value` we do not convert types, if somebody passes a string let them see the assertion

File: projects/kit/components/progress/progress-segmented/progress-segmented.component.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    @tuiDefaultProp(positiveInt, 'Must be positive integer')
+    max = 1;
+
+    @Input()
+    @HostBinding('attr.data-size')
+    size: TuiSizeS = 'm';
+
+    @Input()
+    @tuiDefaultProp()
+    colors: string | ReadonlyArray<string> | null = null;
```
Comment (waterplea) [inline projects/kit/components/progress/progress-segmented/progress-segmented.component.ts:31]: Do we really need `null`? Maybe let's default to `var(--tui-primary)`?

File: projects/demo/src/modules/app/app.routes.ts (hunk 1 above)
Comment (nsbarsukov) [reply projects/demo/src/modules/app/app.routes.ts]: done

File: projects/kit/components/progress/progress-segmented/progress-segmented.component.ts (hunk 2 above)
Comment (nsbarsukov) [reply projects/kit/components/progress/progress-segmented/progress-segmented.component.ts]: fixed

File: projects/kit/components/progress/progress-segmented/progress-segmented.component.ts (hunk 3 above)
Comment (nsbarsukov) [reply projects/kit/components/progress/progress-segmented/progress-segmented.component.ts]: i removed `null`

File: projects/kit/components/progress/progress-segmented/progress-segmented.component.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    @Input()
+    @HostBinding('attr.data-size')
+    size: TuiSizeS = 'm';
+
+    @Input()
+    @tuiDefaultProp()
+    colors: string | ReadonlyArray<string> = 'var(--tui-primary)';
+
+    getActiveColor(index: number = 0): string | null {
+        if (typeof this.colors === 'string') {
```
Comment (waterplea) [inline projects/kit/components/progress/progress-segmented/progress-segmented.component.ts:37]: Ternary :)

**Review by waterplea — APPROVED** [review]

File: projects/kit/components/progress/progress-segmented/progress-segmented.component.ts (hunk 4 above)
Comment (nsbarsukov) [reply projects/kit/components/progress/progress-segmented/progress-segmented.component.ts]: done

**Review by waterplea — APPROVED** [review]

**Pull request merged**